modules/structs/repo_compare.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
```

Comment:
I would rename this file to `repo_diff.go` since it will be reused by other api endpoints